src/core/configuration/Config.ts

```diff
@@ -26,10 +26,15 @@ export enum GameEnv {
   Prod,
 }
 
+export type ServerPlayerConfig = {
+  maxPlayers: number;
+  numPlayerTeams: number | undefined;
+};
+
 export interface ServerConfig {
   turnIntervalMs(): number;
   gameCreationRate(): number;
-  lobbyMaxPlayers(map: GameMapType, mode: GameMode): number;
+  calcPlayerConfig(map: GameMapType, mode: GameMode): ServerPlayerConfig;
```

Comment:
Since this is related to games and not the server per se, maybe this should be LobbyConfig?
```suggestion
  lobbyConfig(map: GameMapType, mode: GameMode): LobbyConfig;
```

Reply:
Yes, that's a better name for it!